src/argilla/server/contexts/accounts.py

```diff
@@ -141,6 +142,31 @@ async def create_user(db: "AsyncSession", user_create: UserCreate) -> User:
     return user
 
 
+async def create_user_with_random_password(
+    db,
+    username: str,
+    first_name: str,
+    workspaces: List[str] = None,
+    role: UserRole = UserRole.annotator,
+    password_length: int = 32,
+) -> User:
+    password = generate_random_password(password_length)
+
+    user_create = UserCreate(
+        first_name=first_name, username=username, role=role, password=password, workspaces=workspaces
+    )
+    return await create_user(db, user_create)
+
+
+def generate_random_password(length: int) -> str:
+    import secrets
+    import string
+
+    alphabet = string.ascii_letters + string.digits
+
+    return "".join([secrets.choice(alphabet) for _ in range(length)])
```

Comment:
You could simply use `secrets.token_urlsafe()` (or even `secrets.token_hex()`):

```suggestion
def generate_random_password() -> str:
    return secrets.token_urlsafe()
```

Some more details:
* Password length is not necessary or if you want you can create a constant for it.
* Remember to move `import secrets` to the imports section.